src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -26,19 +28,26 @@ public static void printLongestWordTop100() throws IOException {
         String contents = new String(Files.readAllBytes(Paths
                 .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
         List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
-
         // TODO 이 부분에 구현한다.
+        words.stream()
+                .filter(word -> word.length() > LONG_WORD_CONDITION)
+                .distinct()
+                .sorted((word1, word2) -> word2.length() - (word1.length()))
+                .limit(WORD_TOP_LIMIT)
+                .map(String::toLowerCase)
+                .forEach(System.out::println);
     }
 
     public static List<Integer> doubleNumbers(List<Integer> numbers) {
         return numbers.stream().map(x -> 2 * x).collect(Collectors.toList());
     }
 
     public static long sumAll(List<Integer> numbers) {
-        return numbers.stream().reduce(0, (x, y) -> x + y);
+        return numbers.stream().reduce(0, Integer::sum);
     }
 
     public static long sumOverThreeAndDouble(List<Integer> numbers) {
-        return 0;
+        return sumAll(doubleNumbers(numbers.stream()
+                .filter(number -> number > 3).collect(Collectors.toList())));
```

Comment:
```suggestion
        return numbers.stream()
                      .filter(v -> v > 3)
                      .map(v -> v * 2)
                      .reduce(0, Math::addExact);
```
이렇게 표현하는게 더 좋을 것 같습니다!